### ISSUE TITLE
Allow to specify more than one context for csolution and cbuild

### ISSUE THREAD

**Issue by ReinhardKeil** [Open-CMSIS-Pack/devtools#823]
For multi-project applications it might be required to build a mixture of build-types.  I suggest therefore to allow to specify multiple `--context` options with the various tools of the CMSIS-Toolbox.  

For example:
```
cbuild my.csolution.yml --context .debug --context hellocm0.release
```
This would build all cproject.yml files with build-type `debug` except the hellocm0.cproject.yml that would use build-type `release`.

By default, I would suggest that that cbuild and csolution uses the first build-type in the list and generates the different files.   To build all different combinations, the option `-c *` would be required.  But details of this option should be discussed.

See also #821

**Comment by ReinhardKeil** [comment]
I suggest that we only add the functionality to `cbuild`.  For `csolution` it is OK to support just one `context` at a time.

**Comment by ReinhardKeil** [comment]
This needs review in after deciding on `context-set`

**Comment by ReinhardKeil** [comment]
See also #896

**Comment by jkrech** [comment]
see CMSIS-Toolbox 2.0.0-dev2

**Comment by jkrech** [comment]
Still waiting for the additional option to replace a context.

**Comment by ReinhardKeil** [comment]
I suggest we close this for now as it seems OK to me.

**Comment by jkrech** [comment]
see: https://github.com/Open-CMSIS-Pack/devtools/pull/1033

Missing was  `--context-replacement`
```
--context .Debug --context myproj1.Release
```
works additive and is not replacing: `myproj1.Debug` and `myproj1.Release` are built

```
--context .Debug --context-replacement myproj1.Release
```
will replace `myproj1.Debug` with `myproj1.Release`

**Comment by jkrech** [comment]
!!! `--context-replacement` is not implemented in CMSIS-Toolbox 2.0.0 
this part of the feature is now captured in #1040
